src/ports/postgres/modules/regress/linear.py_in

```diff
@@ -50,6 +84,13 @@ def linregr_train(schema_madlib, source_table, out_table,
                        independent_varname=independent_varname,
                        source_table=source_table))
 
+        # USING Clause does not support expressions. So modifying to the regular
+        # join clause instead.
+        join_clause = " ON " + ' AND '.join("{source_table}.{col} = {temp_lin_rst}.\""
```

Comment:
I suggest changing this to 
```join_clause = '' if grouping_cols is None else "ON" ...```
Currently the code fails if there are no grouping columns and the `heteroskedasticity_option` is True. We should add a test for this in the dev check as well.

Reply:
Added the above check and Test case as well.